src/libs/bin_decoder/lv_bin_decoder.c

```diff
@@ -403,15 +403,16 @@ lv_result_t lv_bin_decoder_get_area(lv_image_decoder_t * decoder, lv_image_decod
     /*We only support read line by line for now*/
     if(decoded_area->y1 == LV_COORD_MIN) {
         /*Indexed image is converted to ARGB888*/
-        uint32_t len = LV_COLOR_FORMAT_IS_INDEXED(cf) ? sizeof(lv_color32_t) * 8 : bpp;
-        lv_color_format_t cf_decoded = LV_COLOR_FORMAT_IS_INDEXED(cf) ? LV_COLOR_FORMAT_ARGB8888 : cf;
 
-        len = (len * w_px) / 8;
         decoded = decoder_data->decoded_partial;
         if(decoded && decoded->header.w == w_px) {
             /*Use existing one directly*/
         }
         else {
+            if(decoded) {
+                lv_draw_buf_destroy(decoded);
```

Comment:
This issue can happen only in case of tiled images when `decoded` remains valid from the previous tile, right?
If so we can assume that the new `draw_buf` will have the same properties as the previous one. If it's also true having
```c
if(decoded == NULL) decoded = lv_draw_buf_create(w_px, 1, cf_decoded, LV_STRIDE_AUTO);
```
can be better.

@XuNeo could you confirm it? (Neo worked the most on the new image decoders)

Reply:
I added a test that uses that assert. Thanks for the tip. I confirmed that the test does trigger when the bug is enabled.

> This issue can happen only in case of tiled images when decoded remains valid from the previous tile, right?

The issue actually only happens when `decoded` already exists but is _not_ valid. It correctly detects that it is not valid, but forgets to destroy the previous wrong-size existing one.